diff --git a/ReactCommon/react/renderer/core/ShadowNode.cpp b/ReactCommon/react/renderer/core/ShadowNode.cpp
index d762d8b00d497b577082541cfaf3ebe3c0cfebe4..8151f9805ef1c6acf3388d74a6288e0b803d5ce7 100644
--- a/ReactCommon/react/renderer/core/ShadowNode.cpp
+++ b/ReactCommon/react/renderer/core/ShadowNode.cpp
@@ -142,8 +142,9 @@ ShadowNode::Unshared ShadowNode::clone(
     if (fragment.props == ShadowNodeFragment::propsPlaceholder()) {
       // Clone existing `props_` with `family.nativeProps_DEPRECATED` to apply
       // previously set props via `setNativeProps` API.
+      auto nativeProps = family.nativeProps_DEPRECATED;
       auto props = componentDescriptor.cloneProps(
-          propsParserContext, props_, RawProps(*family.nativeProps_DEPRECATED));
+          propsParserContext, props_, RawProps(*nativeProps));
       auto clonedNode = componentDescriptor.cloneShadowNode(
           *this,
           {.props = props,
diff --git a/ReactCommon/react/renderer/core/ShadowNodeFamily.h b/ReactCommon/react/renderer/core/ShadowNodeFamily.h
index 203f895ff7cc16902be5b1e9379ea5bc555e7015..1ecf33588cf7b9d412b1c734df0f8aefe7e5b807 100644
--- a/ReactCommon/react/renderer/core/ShadowNodeFamily.h
+++ b/ReactCommon/react/renderer/core/ShadowNodeFamily.h
@@ -115,7 +115,7 @@ class ShadowNodeFamily final {
    * not used. It is used to backport setNativeProps API from the old
    * architecture and will be removed in the future.
    */
-  mutable std::unique_ptr<folly::dynamic> nativeProps_DEPRECATED;
+  mutable std::shared_ptr<folly::dynamic> nativeProps_DEPRECATED;
 
   /**
    * @return tag for the ShadowNodeFamily.
